extensions/purolator_courier/purplship/providers/purolator_courier/pickup/validate.py

```diff
@@ -86,7 +90,22 @@ def validate_pickup_request(
                 ),
                 FaxNumber=None,
             ),
-            ShipmentSummary=None,
+            ShipmentSummary=ShipmentSummary(
+                ShipmentSummaryDetails = ArrayOfShipmentSummaryDetail(
+                        ShipmentSummaryDetail = [
+                            ShipmentSummaryDetail(
+                                DestinationCode = payload.options.get("ShipmentSummaryDetail")["DestinationCode"],
+                                ModeOfTransport = payload.options.get("ShipmentSummaryDetail")["ModeOfTransport"],
+                                TotalPieces = payload.options.get("ShipmentSummaryDetail")["TotalPieces"],
+                                TotalWeight = Weight(
+                                        Value = payload.options.get("ShipmentSummaryDetail")["WeightValue"],
+                                        WeightUnit = payload.options.get("ShipmentSummaryDetail")["WeightUnit"],
+                                    )
```

Comment:
Same comment as above, `TotalWeight` will be calculated based on the `payload.parcels`.
You can take inspiration from 

```python
packages = Packages(payload.parcels, PackagePresets, required=["weight"])
```

```suggestion
                                TotalWeight = Weight(
                                        Value =packages.weight.LB,
                                        WeightUnit =WeightUnit.LB.value,
                                    )
```